src/backend/base/pyproject.toml

```diff
@@ -211,34 +211,258 @@ pytest-cmd = "pytest -p no:profiling -p no:sugar -p no:xdist -p no:cov -p no:spl
+serpapi = ["google-search-results>=2.4.1,<3.0.0"]
+google-api = ["google-api-python-client==2.154.0"]
+wikipedia = ["wikipedia==1.4.0"]
+fake-useragent = ["fake-useragent==1.5.1"]
+duckduckgo = ["duckduckgo_search==7.2.1"]
+yfinance = ["yfinance==0.2.50"]
+wolframalpha = ["wolframalpha==5.1.3"]
+
+# Individual utilities
+pyarrow = ["pyarrow==19.0.0"]
+fastavro = ["fastavro==1.9.7"]
+gitpython = ["GitPython==3.1.43"]
+nltk = ["nltk==3.9.1"]
+lark = ["lark==1.2.2"]
+
+# Additional dependencies from requirements
+huggingface = ["huggingface-hub[inference]>=0.23.2,<1.0.0"]
+metaphor = ["metaphor-python==0.1.23"]
+metal = ["metal_sdk==2.5.1"]
+markup = ["MarkupSafe==3.0.2"]
+aws = ["boto3==1.34.162"]
+numexpr = ["numexpr==2.10.2"]
+qianfan = ["qianfan==0.3.5"]
+pgvector = ["pgvector==0.3.6"]
+litellm = ["litellm==1.60.2"]
+zep = ["zep-python==2.0.2"]
+youtube = ["youtube-transcript-api==0.6.3"]
+markdown = ["Markdown==3.7"]
+kubernetes = ["kubernetes==31.0.0"]
+json-repair = ["json_repair==0.30.3"]
+astra = ["astra-assistants[tools]~=2.2.9"]
+composio = [
+    "composio-langchain==0.7.1",
+    "composio-core==0.7.1"
 ]
+ragstack = ["ragstack-ai-knowledge-store==0.2.1"]
+opensearch = ["opensearch-py==2.8.0"]
+atlassian = ["atlassian-python-api==3.41.16"]
+mem0 = ["mem0ai==0.1.34"]
+needle = ["needle-python>=0.4.0"]
+sseclient = ["sseclient-py==1.8.0"]
+openinference = ["openinference-instrumentation-langchain>=0.1.29"]
+mcp = ["mcp>=0.9.1"]
+ag2 = ["ag2>=0.1.0"]
+scrapegraph = ["scrapegraph-py>=1.12.0"]
+pydantic-ai = ["pydantic-ai>=0.0.19"]
+apify = ["apify-client>=1.8.1"]
+spider = ["spider-client>=0.0.27,<1.0.0"]
+
+# Development dependencies
+dev = [
+    "types-redis>=4.6.0.5",
+    "ipykernel>=6.29.0",
+    "mypy>=1.11.0",
+    "ruff>=0.9.1,<0.10",
+    "httpx>=0.27.0",
+    "pytest>=8.2.0",
+    "types-requests>=2.32.0",
+    "requests>=2.32.0",
+    "pytest-cov>=5.0.0",
+    "pandas-stubs>=2.1.4.231227",
+    "types-pillow>=10.2.0.20240213",
+    "types-pyyaml>=6.0.12.8",
+    "types-python-jose>=3.3.4.8",
+    "types-passlib>=1.7.7.13",
+    "pytest-mock>=3.14.0",
+    "pytest-xdist>=3.6.0",
+    "types-pywin32>=306.0.0.4",
+    "types-google-cloud-ndb>=2.2.0.0",
+    "pytest-sugar>=1.0.0",
+    "respx>=0.21.1",
+    "pytest-instafail>=0.5.0",
+    "pytest-asyncio>=0.23.0",
+    "pytest-profiling>=1.7.0",
+    "pre-commit>=3.7.0",
+    "vulture>=2.11",
+    "dictdiffer>=0.9.0",
+    "pytest-split>=0.9.0",
+    "pytest-flakefinder>=1.1.0",
+    "types-markdown>=3.7.0.20240822",
+    "packaging>=24.1,<25.0",
+    "asgi-lifespan>=2.1.0",
+    "pytest-github-actions-annotate-failures>=0.2.0",
+    "pytest-codspeed>=3.0.0",
+    "blockbuster>=1.5.20,<1.6",
+    "types-aiofiles>=24.1.0.20240626",
+    "codeflash>=0.8.4",
+    "hypothesis>=6.123.17",
+    "locust>=2.32.9"
+]
+
+# Complete installation
 local = [
-    "llama-cpp-python>=0.2.0",
-    "sentence-transformers>=2.0.0",
-    "ctransformers>=0.2"
+    "langflow-base[deploy]",
```

Comment:
IIRC, `extras` don't compose.
Does it work ? Are the `extras` present in the package metadata ?

Reply:
https://github.com/astral-sh/uv/issues/4762#issuecomment-2207423858

I got this from here.

Reply:
I tested and it works!
`uv` is better than `poetry` for this !
Sorry for the noise.